### ISSUE TITLE
Embed component video attachment file got error AccessDenied when showed

### ISSUE THREAD

**Issue by chenhuanghui** [NotionX/react-notion-x#520]
#### Description
I have file video attachment on notion page as embed component, and using react-notion-x to render content.
But I got error as below:

```
<Error>
<Code>AccessDenied</Code>
<Message>Access Denied</Message>
<RequestId>RESB8AZZZ1BMMPYY</RequestId>
<HostId>n8IIFPpQZblmwRVv2QUpx/RfULLwGXauji9s52oaKfSlhn1O/pdTb7umphLSQd1Mk1yZ4+FwDf8=</HostId>
</Error>
```

#### Notion Test Page ID
This is my notion page ID: https://www.notion.so/casavn/test-embed-page-046f035b59f346d0bdf59f1175f94186?pvs=4

React-Notion-X render
https://www.casaholding.vn/046f035b59f346d0bdf59f1175f94186

**Comment by 0xFloyd** [comment]
Im also getting this error, didnt in the past

**Comment by vader1359** [comment]
Anyone figured out a work around this?

**Comment by brunomorency** [comment]
Notion now exposes uploaded content through expiring links. You either need to cache those filees, fetch a fresh link from Notion every time your page loads or, in the Notion UI, embed files as external links rather than uploading them on Notion.

**Comment by jpprakash4260** [comment]
Does anyone know any alternative way or method to bypass these expiring links?

**Comment by GridexX** [comment]
There should be an update on the Notion side, the video is correctly showing up on my side

**Comment by dennisrcao** [comment]
Yup same issue here, ive got an uploaded video to a notion page (thats been published) when I use the API to read off of it then re-render the page, everything loads except for the embedded videos. And on the DOM of the re-rendered page there is a `<video> `component but when I go to the value of the `src` parameter (which is the url it's trying to load ).
![image](https://github.com/user-attachments/assets/bba81bad-806e-4e5a-bd3c-12d577c50230)

 I get this error.
Essentially the same issue that the OP is saying.

**Comment by kyeah** [comment]
It seems possible to retrieve a new URL when needed by providing the following format to `notion.so/signed`:

```
https://notion.so/signed/${encodedUrl}?table=block&id=${videoBlockId}&spaceId=${spaceId}
```

The built-in notion-x CSS styling seems like it needs updates to support this method (or maybe my site is applying some custom styling.) 

Regardless, I was able to get the videos to show up by editing the block video sources before passing it to the NotionRenderer.

```js
export default function updateVideoUrl(block) {
  if (block.type != "video") {
    return;
  }

  const s3Url = block.properties.source[0][0];
  if (s3Url.includes("notion.so/signed")) {
    return;
  }

  const encodedUrl = encodeURIComponent(s3Url);
  block.properties.source[0][0] = `https://notion.so/signed/${encodedUrl}?table=block&id=${block.id}&spaceId=${block.space_id}`;
}
```

See this branch: https://github.com/spacetypeco/blogs/compare/main...mapVideoUrl

I didn't test this thoroughly so YMMV. This was with a public page.

**Comment by worgho2** [comment]
Thanks @kyeah for mentioning the open PR, by the way a possible workaround is to overrive the existing `addSignedUrls` function. Check this out: https://github.com/NotionX/react-notion-x/issues/580#issuecomment-2542823817
